### PR TITLE
Add documentation about CLI flags and environment variables

### LINKED ISSUE #1706: Document CLI flags and environment variables
### What should be done?

Our documentation should provide more information than just `--help` flag.

Most importantly:
* the `--mode` flag requires some explanation in user documentation, not only in CONTRIBUTING.md;
* TLS flags should be explained.

_Related: https://ferretdb.slack.com/archives/C049W6E01E0/p1672187174464469_

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #1706.

## Readiness checklist

* [ ] I added unit tests for new functionality or bug fixes.
* [ ] I added integration tests for new functionality or bug fixes.
* [ ] I added compatibility tests for new functionality or bug fixes.
* [ ] I made spot refactorings.
* [x] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by rumyantseva — COMMENTED** [review]
Great job describing all the flags! 👍 

I'd say we need a check from @Fashander to ensure our docs sound user-friendly and easy to read. But to me, it looks really great! I added a few suggestions to clarify a couple of things.

File: website/docs/operation_modes.md
Comment (noisersup) [reply website/docs/operation_modes.md:48]: good idea!

Comment (noisersup) [reply website/docs/operation_modes.md]: done, couldn't find any workin collsize solution quickly, so used `datasize` stub as an example. WDYT?

**Review by w84thesun — APPROVED** [review]
LGTM!

File: website/docs/flags.md
Comment (Fashander) [inline website/docs/flags.md:6]:
```suggestion
FerretDB provides numerous configuration flags that you can customize to suit your needs and environment.
```

**Review by rumyantseva — DISMISSED** [review]
LGTM

File: website/docs/flags.md
Comment (Fashander) [inline website/docs/flags.md:33]: I think the link should be `./operation_modes.md`.

File: website/docs/flags.md
Comment (Fashander) [inline website/docs/flags.md:34]: The same for the one here too. `./operation_modes.md#proxy`

File: website/docs/operation_modes.md
Comment (Fashander) [inline website/docs/operation_modes.md:6]:
```suggestion
To simplify the development and debugging of FerretDB, we support different operation modes.
```

File: website/docs/operation_modes.md
Comment (Fashander) [inline website/docs/operation_modes.md:8]:
```suggestion
They are useful for testing, debugging, or bug reporting.
```

**Review by Fashander — COMMENTED** [review]
Good job on this! 👍🏽 Just a few minor suggestions.

File: website/docs/flags.md
Comment (noisersup) [reply website/docs/flags.md:33]: Both of them are in root directory, so functionally it doesn't make any change, do we prefer relative paths?

File: website/docs/flags.md
Comment (b1ron) [reply website/docs/flags.md:36]: My complete misunderstanding I will resolve :D

**Review by b1ron — APPROVED** [review]
Looks good!

File: website/docs/flags.md
Comment (noisersup) [reply website/docs/flags.md:36]: Yeah, you can access them before merging by going to Preview page created by vercel bot

**Review by w84thesun — APPROVED** [review]
👍

**Review by rumyantseva — APPROVED** [review]
LGTM

File: website/docs/flags.md
Comment (Fashander) [reply website/docs/flags.md:33]: Did you try out the links? Running `task docs-dev` fails because of the links.

Comment (noisersup) [reply website/docs/flags.md:33]: Weird
```sh
 task docs-dev
task: [docs-dev] docker compose run --rm --service-ports docusaurus-docs start --host=0.0.0.0
[INFO] Starting the development server...
[SUCCESS] Docusaurus website is running at: http://localhost:3000/

✔ Client
  Compiled successfully in 8.65s

client (webpack 5.74.0) compiled successfully

✔ Client
  Compiled successfully in 135.23ms

client (webpack 5.74.0) compiled successfully
```

Comment (noisersup) [reply website/docs/flags.md:33]: Did you try `task env-down init`?

Comment (Fashander) [reply website/docs/flags.md:33]: Yea. I did. Really weird. But I guess it should work anyways.

**Review by Fashander — APPROVED** [review]

**Comment by noisersup** [comment]
Because of change in `CONTRIBUTING.md` @AlekSi approval is necessary. [CODEOWNERS](https://github.com/FerretDB/FerretDB/blob/main/.github/CODEOWNERS)

**Comment by rumyantseva** [comment]
@noisersup let's ask @ptrfarkas to help us 😉

**Review by ptrfarkas — APPROVED** [review]

**Pull request merged**
